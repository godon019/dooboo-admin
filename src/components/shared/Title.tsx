import React, { Component } from 'react';

import styled from 'styled-components';

const AdminTitle = styled.div`
  /* margin-bottom: 3rem; */
`;
const LogoImgArea = styled.div`
  display: flex;
  align-items: center;
  padding-bottom: 0.2rem;
`;
const TitleDesc = styled.span<{ larger: boolean }>`
  font-size: ${(props) => (props.larger ? '1.5rem' : '.8rem')};
  padding: 0 0.3rem;
  color: #cbcccf;
`;
const LogoImg = styled.img<{ larger: boolean }>`
  :hover {
    cursor: pointer;
  }
  width: ${(props) => (props.larger ? '9rem' : '5rem')};
  height: auto;
`;
const Text = styled.div`
  font-size: 1.6rem;
  font-weight: bold;
`;

type TitleProps = {
  text?: string;
  src?: any;
  larger?: boolean;
  onLogoClick?: () => void;
};

const Title: React.FC<TitleProps> = ({
  text,
  src = 'https://dooboolab.com/8a19465015cf5126862d9e850962f835.png',
  larger,
  onLogoClick,
}) => {
  return (
    <AdminTitle>
      <LogoImgArea>
        <LogoImg src={src} larger={larger} onClick={onLogoClick} />
        <TitleDesc larger={larger}>{`| Admin`}</TitleDesc>
      </LogoImgArea>
      <Text>{text}</Text>
    </AdminTitle>
  );
};

export default Title;
