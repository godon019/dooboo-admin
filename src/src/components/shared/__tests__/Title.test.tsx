import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import Title from '../Title';
import { render, fireEvent, getByTestId } from '@testing-library/react';

let props: any;
let component: React.ReactElement;
let testingLib: any;

describe('[Title] render', () => {
  beforeEach(() => {
    props = {
      history: {
        goBack: jest.fn(),
      },
    };
    component = (
      <Title />
    );
  });

  it('renders without crashing', () => {
    const rendered: ReactTestRendererJSON = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      // fireEvent(testingLib.getByTestId('btn'), 'click');
      // expect(cnt).toBe(2);
    });
  });
});
