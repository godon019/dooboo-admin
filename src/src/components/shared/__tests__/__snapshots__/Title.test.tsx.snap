// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Title] render renders without crashing 1`] = `
<div
  className="sc-bdVaJa dzGXzv"
>
  <div
    className="sc-bwzfXH emXnOW"
  >
    <img
      className="sc-bxivhb dcjKSD"
    />
    <span
      className="sc-htpNat bGakfH"
    >
      | Admin Page
    </span>
  </div>
  <div
    className="sc-ifAKCX cSmKN"
  />
</div>
`;
